motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -695,6 +744,116 @@ BOOL Ros_MotionServer_StopMotion(Controller* controller)
 }
 
 
+//-----------------------------------------------------------------------
+// Sets servo power to ON or OFF
+//-----------------------------------------------------------------------
+BOOL Ros_MotionServer_ServoPower(Controller* controller, int servoOnOff)
+{
+	MP_SERVO_POWER_SEND_DATA sServoData;
+	MP_STD_RSP_DATA stdRespData;
+	int ret;
+	
+	if (servoOnOff == OFF)
+		Ros_MotionServer_StopMotion(controller);
+
+	printf("Setting servo power: %d\n", servoOnOff);
+	memset(&sServoData, 0x00, sizeof(sServoData));
+	memset(&stdRespData, 0x00, sizeof(stdRespData));
+	sServoData.sServoPower = servoOnOff;
+
+	ret = mpSetServoPower(&sServoData, &stdRespData);
+	if( (ret == 0) && (stdRespData.err_no == 0) )
+	{
+		// wait for confirmation
+		int checkCount;
+		for(checkCount=0; checkCount<MOTION_START_TIMEOUT; checkCount+=MOTION_START_CHECK_PERIOD)
+		{
+			// Update status
+			Ros_Controller_StatusUpdate(controller);
+		
+			if(Ros_Controller_IsServoOn(controller) == servoOnOff)
+				break;
+			
+			mpTaskDelay(MOTION_START_CHECK_PERIOD);
+		}		
+	}
+	else
+	{
+		char errMsg[ERROR_MSG_MAX_SIZE];
+		memset(errMsg, 0x00, ERROR_MSG_MAX_SIZE);
+		Ros_Controller_ErrNo_ToString(stdRespData.err_no, errMsg, ERROR_MSG_MAX_SIZE);
+		printf("Can't turn off servo because: %s\r\n", errMsg);
+	}
+	
+	// Update status
+	Ros_Controller_StatusUpdate(controller);
+	return Ros_Controller_IsServoOn(controller) == servoOnOff;
+}
+
+BOOL Ros_MotionServer_ResetAlarm(Controller* controller)
+{
+	int ret, i;
+	BOOL returnBoolean;
+	MP_ALARM_STATUS_RSP_DATA alarmstatus;
+	MP_STD_RSP_DATA responseData;
+	
+	returnBoolean = TRUE;
+
+	ret = mpGetAlarmStatus(&alarmstatus);
+	if( ret != 0 ) 
+	{
+		printf("Could not get alarm status\n");
+		//Ignore this error.  Continue to try and clear the alarm.
+	}
+    
+	if (alarmstatus.sIsAlarm & 0x02) //alarm is active

```

Comment:
Could we perhaps avoid the magic nr (`0x02`)? A local `#define` or `const` would be nicer.